specification/maps/data-plane/Microsoft.Maps/Tileset/preview/2022-01-01-preview/tileset.json

```diff
@@ -594,8 +711,100 @@
           "items": {
             "type": "number"
           }
+        },
+        "tilesetRecipe": {
+          "description": "Tileset recipe which is used to build the tileset.",
+          "$ref": "#/definitions/TilesetRecipe"
+        }
+      }
+    },
+    "TilesetRecipe": {
+      "description": "A tileset recipe object. Please refer to [Tileset Recipe Specification]() for details.",
+      "type": "object",
+      "required": [
+        "version",
+        "layers"
+      ],
+      "properties": {
+        "version": {
+          "description": "The version of Azure Maps Tiling service recipe reference that the recipe uses.",
+          "type": "integer"
+        },
+        "layers": {
+          "description": "The names of the layers and their configuration options.",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/LayerObject"
+          }
+        }
+      }
+    },
+    "LayerObject": {
+      "description": "Layer object for tileset recipe.",
+      "type": "object",
+      "required": [
+        "features",
+        "minzoom",
+        "maxzoom"
+      ],
+      "properties": {
+        "features": {
+          "description": "Specifies the output on a per feature basis.",
+          "$ref": "#/definitions/Feature"
+        },
+        "minzoom": {
+          "description": "The lowest zoom level for the layer to build.",
+          "type": "integer"
+        },
+        "maxzoom": {
+          "description": "The highest zoom level for the layer to build.",
+          "type": "integer"
+        }
+      }
+    },
+    "Feature": {
+      "description": "Feature configuration for tileset recipe.",
+      "type": "object",
+      "required": [
+        "featureClasses"
+      ],
+      "properties": {
+        "featureClasses": {
+          "description": "The list of featureClasses that will be used to build the layer.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "filter": {
+          "description": "Removes features that do not match a pattern.",
+          "type": "object"
```

Comment:
This is a very vague type. Is this a dictionary?

Reply:
This is actually a Json array which represent a mapbox expression, the element of the array can be another expression. So I changed it to be an array of object instead.